src/rgw/driver/rados/rgw_bucket.cc

```diff
@@ -1337,13 +1337,26 @@ static int bucket_stats(rgw::sal::Driver* driver,
   }
 
   // bucket notifications
-  RGWPubSub ps(driver, tenant_name);
   rgw_pubsub_bucket_topics result;
-  const RGWPubSub::Bucket b(ps, bucket.get());
-  ret = b.get_topics(dpp, result, y);
-  if (ret < 0 && ret != -ENOENT) {
-    cerr << "ERROR: could not get topics: " << cpp_strerror(-ret) << std::endl;
-    return -ret;
+  if (static_cast<rgw::sal::RadosStore* const>(driver)
```

Comment:
why do we need to cast to `RadosStore`? this code is inside the rados driver

Reply:
also, this "if" statement mixes 2 things that should be separate - the new data model and data syncing

Reply:
yes the function is only inside the `RadosStore`, thats why had to cast...
mixes 2 things ? i am calling a function to check whether to use new model/multisite syncing!
If you think the name is confusing.. we could add the feature flag name as v2... but still its a zonegroup feature... this way we do not need 2 flags... one for new model and one for syncing !

Reply:
please don't cast to RadosStore. this won't work with other backends, or when there are filters in front of the RadosStore (an example filter is the work-in-progress data caching layer in https://github.com/ceph/ceph/pull/52444)

see https://github.com/ceph/ceph/blob/de5d8cae/src/rgw/rgw_op.cc#L4340-L4342 for an example of a zonegroup feature check that doesn't depend on RadosStore. you can pass `const RGWZoneGroup&` into these functions for the `zonegroup.supports()` call

Reply:
i thought about it initially, but `RGWZoneGroup` is not available at lot of places where i want..
I  need to access the feature flag from rgw_notify.cc and rgw_pubsub.cc and i do not have access to RGWZoneGroup.
do you know any other way to get `RGWZoneGroup` other than `s->penv.site->get_zonegroup()` or via `rgw::read_zonegroup()`

Reply:
so i think i could add a function supports_feature to [Class ZoneGroup](https://github.com/ceph/ceph/blob/main/src/rgw/rgw_sal.h#L1404), and it internally calls the `RGWZoneGroup->supports` and getting ZoneGroup() instance is easy `driver->get_zone()->get_zonegroup();`
so i could replace all the calls with `driver->get_zone()->get_zonegroup()->supports_feature('<feature_name>')`